module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp

```diff
@@ -131,66 +233,108 @@ namespace module::localisation {
                     state_cov.topLeftCorner(2, 2) = pos_cov.matrix();
                     state_cov(2, 2)               = s.heading_var;
                     cov.push_back(state_cov);
-                    filter.set_state(hfw_state_vec, state_cov);
                 }
+                filter.set_state(states, cov);
             });
 
-        on<Configuration>("RobotParticleLocalisation.yaml").then([this](const Configuration& config) {
+        on<Configuration>("RobotParticleLocalisation.yaml").then([this](const Configuration& cfg) {
             // Use configuration here from file RobotParticleLocalisation.yaml
-            filter.model.processNoiseDiagonal = config["process_noise_diagonal"].as<Expression>();
-            filter.model.n_rogues             = config["n_rogues"].as<int>();
-            filter.model.resetRange           = config["reset_range"].as<Expression>();
-            filter.model.n_particles          = config["n_particles"].as<int>();
-            draw_particles                    = config["draw_particles"].as<int>();
-
-            Eigen::Vector3d start_state = config["start_state"].as<Expression>();
-            // TODO: This variable is not used. Probably remove it
-            /* Eigen::Vector3d start_variance = config["start_variance"].as<Expression>(); */
-
-            auto reset = std::make_unique<ResetRobotHypotheses>();
-            ResetRobotHypotheses::Self leftSide;
-            // Start on goal line
-            leftSide.position     = Eigen::Vector2d(start_state.x(), start_state.y());
-            leftSide.position_cov = Eigen::Vector2d::Constant(0.5).asDiagonal();
-            leftSide.heading      = start_state.z();
-            leftSide.heading_var  = 0.005;
-
-            reset->hypotheses.push_back(leftSide);
-            ResetRobotHypotheses::Self rightSide;
-            // Start on goal line
-            rightSide.position     = Eigen::Vector2d(start_state.x(), -start_state.y());
-            rightSide.position_cov = Eigen::Vector2d::Constant(0.5).asDiagonal();
-            rightSide.heading      = -start_state.z();
-            rightSide.heading_var  = 0.005;
-
-            reset->hypotheses.push_back(rightSide);
-            emit<Scope::DELAY>(reset, std::chrono::seconds(1));
+            filter.model.processNoiseDiagonal = cfg["process_noise_diagonal"].as<Expression>();
+            filter.model.n_rogues             = cfg["n_rogues"].as<int>();
+            filter.model.resetRange           = cfg["reset_range"].as<Expression>();
+            filter.model.n_particles          = cfg["n_particles"].as<int>();
+
+            config.debug = cfg["debug"].as<bool>();
+
+            config.start_variance = cfg["start_variance"].as<Expression>();
+        });
+
+        on<Startup, With<FieldDescription>>().then([this](const FieldDescription& fd) {
+            // Left side penalty mark
+            config.start_state.emplace_back((-fd.dimensions.field_length / 2.0) + fd.dimensions.penalty_mark_distance,
```

Comment:
It would be truly amazing if we could get a diagram in nubook describing these. My brian hurts thinking about it, even though I can see it's not that complex lol

Reply:
If I remember correctly what happened here is that when we reinitialised the particle filter we put a bunch of particles in all the possible locations that the robot could start in.
Then the particles where the robot actually is should in theory be reinforced while the others should be diminished.

Reply:
working on a diagram/a better way to init the particles, that will come in another PR though